src/config.py

```diff
@@ -320,11 +320,22 @@ def _build_destination(
                 f' "{dest_config["ref"]}" defined in config'
             ) from e
 
+        try:
+            request_timeout = dest_config["timeout"]
+            if request_timeout < 1:
+                raise ValueError("Request timeout for DuneClient cannot be < 1")
+            elif type(request_timeout) != int:
+                raise ValueError("Request timeout must be int")
+        except KeyError:
+            log.debug("Request timeout not set: defaulting to 10")
```

Comment:
Shouldn't this move into the dune case branch given that it's not relevant for other data sources?
```suggestion
            log.debug("Dune request timeout not set: defaulting to 10")
```